src/cmd/build.rs

```diff
@@ -293,6 +284,63 @@ fn optimize_wasm(crate_metadata: &CrateMetadata) -> Result<OptimizationResult> {
     })
 }
 
+/// Optimizes the Wasm supplied as `wasm` using the `binaryen-rs` dependency.
+///
+/// If successful, the optimized Wasm is returned as a `Vec<u8>`.
+#[cfg(feature = "wasm-opt-unavailable")]
+fn get_optimized_wasm(_: &CrateMetadata, _: &Path, wasm: &[u8]) -> Result<Vec<u8>> {
+    let codegen_config = binaryen::CodegenConfig {
+        // execute -O3 optimization passes (spends potentially a lot of time optimizing)
+        optimization_level: 3,
+        // the default
+        shrink_level: 1,
+        // the default
+        debug_info: false,
+    };
+    let mut module = binaryen::Module::read(&wasm)
+        .map_err(|_| anyhow::anyhow!("binaryen failed to read file content"))?;
+    module.optimize(&codegen_config);
+    Ok(module.write())
+}
+
+/// Optimizes the Wasm supplied as `crate_metadata.dest_wasm` using
+/// the `wasm-opt` binary.
+///
+/// If successful, the optimized Wasm file is created under `optimized`
+/// and returned as a `Vec<u8>`.
+#[cfg(not(feature = "wasm-opt-unavailable"))]
+fn get_optimized_wasm(
```

Comment:
Maybe `do_optimization` or similar, `get` gives the impression that it is just accessing some existing resource rather than performing a process.